token/services/identity/msp/x509/lm.go

```diff
@@ -333,6 +332,24 @@ func (lm *LocalMembership) getResolver(label string) *common.Resolver {
 	return nil
 }
 
+func (lm *LocalMembership) keyStorePath(translatedPath string) (keyStorePath string) {
+	if lm.ignoreVerifyOnlyWallet {
```

Comment:
Maybe it's easier if we invert the logic:
```
if !lm.ignoreVerifyOnlyWallet {
 return ""
}

path := filepath.Join(translatedPath, KeystoreFullFolder)
if _, err := os.Stat(path); err == nil {
 return path
}

path = filepath.Join(translatedPath, "msp", KeystoreFullFolder)
if _, err := os.Stat(path); err == nil {
 return path
}

return ""

Reply:
👍